aci/resource_aci_dhcpoptionpol.go

```diff
@@ -99,26 +95,25 @@ func getRemoteDHCPOptionPolicy(client *client.Client, dn string) (*models.DHCPOp
 	if err != nil {
 		return nil, err
 	}
-
 	dhcpOptionPol := models.DHCPOptionPolicyFromContainer(dhcpOptionPolCont)
-
 	if dhcpOptionPol.DistinguishedName == "" {
-		return nil, fmt.Errorf("DHCPOptionPolicy %s not found", dhcpOptionPol.DistinguishedName)
+		return nil, fmt.Errorf("DHCPOptionPolicy %s not found", dn)
 	}
-
 	return dhcpOptionPol, nil
 }
 
-func getRemoteDHCPOptionFromDHCPOptionPolicy(client *client.Client, dn string) (*models.DHCPOption, error) {
-	dhcpOptionCont, err := client.Get(dn)
+func getRemoteDHCPOptionsFromDHCPOptionPolicy(client *client.Client, dn string) ([]*models.DHCPOption, error) {
+	dhcpOptionCont, err := client.GetViaURL(fmt.Sprintf("%s/%s/%s.json", "/api/node/class", dn, "dhcpOption"))
 	if err != nil {
 		return nil, err
 	}
 
-	dhcpOption := models.DHCPOptionFromContainer(dhcpOptionCont)
+	dhcpOption := models.DHCPOptionListFromContainer(dhcpOptionCont)
 
-	if dhcpOption.DistinguishedName == "" {
-		return nil, fmt.Errorf("DHCPOption %s not found", dhcpOption.DistinguishedName)
+	for _, option := range dhcpOption {
+		if option.DistinguishedName == "" {
+			return nil, fmt.Errorf("DHCPOption %s not found", dn)
```

Comment:
dn is the parent here right? this condition and error message does not really make sense.
Shouldn't it return that error when no DHCPOption are returned? so the lengh of the list is probably better.

Reply:
Yes. I did change this part quite a bit. User should be able to add dhcp option policy with no options specified as well.